salt/orch/update.sls

```diff
@@ -138,6 +139,28 @@ pre-orchestration-migration:
     - require:
       - salt: {{ master_id }}-post-start-services
 
+# Remove the now defuct caasp_fqdn grain
+{{ master_id }}-remove-caasp-fqdn-grain:
+  salt.function:
+    - tgt: {{ master_id }}
+    - name: grains.delval
+    - arg:
+      - caasp_fqdn
+      - {{ master_id }}
+    - require:
+      - salt: {{ master_id }}-reboot-needed-grain
+
+# Ensure the node is marked as finished upgrading
+{{ master_id }}-remove-update-grain:
+  salt.function:
+    - tgt: {{ master_id }}
+    - name: grains.setval
+    - arg:
+      - update_in_progress
+      - false
+    - require:
+      - salt: {{ master_id }}-remove-caasp-fqdn-grain
```

Comment:
Since we still need to work with `masters` (check the `super_master` at the end) we cannot release this grain yet. This would add a window in which the `update-etc-hosts` orchestration can get in between.

1. `grains.setval` -> `update_in_progress: false`
2. Some machine restarts, triggers `update-etc-hosts` orchestration
3. A state.sls is triggered in what we'll call the `super_master` (the first master of the list)
4. We reach https://github.com/kubic-project/salt/blob/dfd3b8a6a65c7d969466b09a1f20536a525ae42a/salt/orch/update.sls#L247-L271, and the process fails because "state.sls is running as PID...".
5. The update orchestration fails.

This has been one of the reasons of why the `orch.update` is so unstable and is causing so many problems. This grain will be released here: https://github.com/kubic-project/salt/blob/dfd3b8a6a65c7d969466b09a1f20536a525ae42a/salt/orch/update.sls#L273-L283, so this entire `{{ master_id }}-remove-update-grain` can be removed.

The idea is that the `update-etc-hosts` orchestration will only render the machines we have actually finished working with when in the middle of an update orchestration.